apiv2/internal/server/utils.go

```diff
@@ -81,6 +94,40 @@ func SafeUint64ToUint32(value uint64) (uint32, error) {
 	return uint32(value), nil
 }
 
+// TruncateUint64ToUint32 truncates the lower bits of a uint64 to fit into a uint32.
+func TruncateUint64ToUint32(value uint64) uint32 {
+	return uint32(value & math.MaxUint32) //nolint:gosec // Mask the lower 32 bits.
+}
+
+// SafeInt64ToUint64 safely converts a int64 to a uint64.
+func SafeInt64ToUint64(value int64) (uint64, error) {
+	if value < 0 {
+		return 0, errors.New("cannot convert a negative int64 to a uint64")
+	}
+
+	res := uint64(value)
+	if int64(res) != value { //nolint:gosec // no risk of overflow
+		return 0, errors.New("int exceeds uint64 limit")
+	}
+	return res, nil
+}
+
+// SafeUint32Toint32 safely converts a uint32 to a int32.
+func SafeUint32Toint32(value uint32) (int32, error) {
+	if value > math.MaxInt32 {
+		return 0, errors.New("value exceeds int32 range")
+	}
+	return int32(value), nil
+}
+
+// SafeUint64ToInt64 safely converts a uint64 to an int64.
+func SafeUint64ToInt64(value uint64) (int64, error) {
+	if value > math.MaxInt64 {
+		return 0, errors.New("value exceeds int64 range")
+	}
+	return int64(value), nil
+}
```

Comment:
Please, check, some of these are already in Inventory/pkg/util/util.go

Reply:
Done.